kernel/src/sched.rs

```diff
@@ -623,8 +623,12 @@ impl Kernel {
                     // why and handle the process as appropriate.
                     match context_switch_reason {
                         Some(ContextSwitchReason::Fault) => {
-                            // Let process deal with it as appropriate.
-                            process.set_fault_state();
+                            // The app faulted, check if the chip wants to
+                            // handle the fault.
+                            if platform.process_app_fault(process).is_err() {
+                                // Let process deal with it as appropriate.
+                                process.set_fault_state();
+                            }
```

Comment:
Possibly more of a style question than a correctness one, should this include a `continue` here in the case where the exception was handled by the platform? We include (effectively unused, since the match is the last statement in this loop) `continue`s in all of the match/branches below. I _think_ the intent is to be specific about 'this case means keep running this process (`continue`) or stop (`break`).